.github/CODEOWNERS

```diff
@@ -457,7 +457,7 @@ packages/kbn-kibana-manifest-schema @elastic/kibana-operations
 src/plugins/kibana_overview @elastic/appex-sharedux
 src/plugins/kibana_react @elastic/appex-sharedux
 src/plugins/kibana_usage_collection @elastic/kibana-core
-src/plugins/kibana_utils @elastic/kibana-app-services
+src/plugins/kibana_utils @elastic/kibana-core
```

Comment:
Feels like this should be shared UX. I don't think anyone on core was involved with the vast majority of the code in this plugin.

Reply:
I second this 🙂

Reply:
Shared UX also was not involved with the vast majority of the code in this plugin, does not feel right to assign it to Shared UX.

Reply:
Okay I went ahead and did an audit of `kibana_utils`. I think the problem is that most of the domains here which were historically app architecture/services have since been split mostly among shared ux, visualizations, data discovery, and presentation teams.

Giving the plugin a quick glance, these are the domains I'm seeing.

public:
- core (createStartServicesGetter)
- history
- render complete
- resize checker
- state management
- state sync
- storage
- theme
- ui

server:
- KbnServerError & related utilities

common:
- KbnError and other error utilities
- persistable state
- state containers
- state management & url helpers
- a handful of other small utilities which appear largely used by data/discover plugins

So, given these domains:
- AFAIK, state containers, state management / sync, history, storage, theme, ui are all Shared UX concerns... are they not?
- KbnError-related items are mostly from the data plugin and consumed by downstream services, which would be Data Discovery or Vis teams?
- Render/resize checkers and persistable state are (I think) mostly for embeddables -> presentation team.
- createStartServicesGetter was originally written for expressions -> visualizations team.

After reviewing everything, I truly don't think the core team contributed a single one of the utilities to this plugin. These are all helpers to use when building kibana apps, they are not foundational platform-level constructs.

So if Shared UX doesn't want to take ownership of everything here, then maybe the solution is to break up the plugin. Most of these items are only used in 1 or 2 places and could easily be copied to where they are consumed. For the ones that are more widespread, we could assign codeowners to those specific subdirectories based on domain.

Regardless of the direction we take, I'm not comfortable with making core owners of this just because it happens to be a generic bucket of utilities that are difficult to place under a single domain.

Reply:
@vadimkibana The previous owner was app-services, which was comprised of most of the folks who now form Shared UX.  You can only say "Shared UX wasn't involved" because Shared UX _didn't exist_.  The commit history for `kibana_utils` is a who's who of App Services team members or experts, where a number of systems were written by you, @Dosant, @ppisljar and a few others.

Honestly, I don't want this to be a plugin anymore.  I want to see this code audited, broken into packages and owned by teams who use them.  But frankly, you are one of the subject matter experts on a lot of the code in this plugin, if not one of "the owners".  Just tossing it to someone else isn't appropriate.

I think this plugin _should_ be owned by Shared UX, and we should audit and split it up before finding appropriate owners for those packages.

_...and thank you, @lukeelmers, for doing the due diligence with this initial audit._

Reply:
+1 for splitting up this plugin. Maybe adding code ownership to sub-folders would be the best initial step.

> After reviewing everything, I truly don't think the core team contributed a single one of the utilities to this plugin.

I don't think that it should be determined by which team contributed the code, but we are looking for the best natural long-term owner.

---

From Luke's list the items that are natural for SharedUX to break up into packages and maintain going forward:

- render complete - used in functional tests and reporting.
- theme - deprecated, SharedUX can see through its removal.
- ui - TypeScript types used in UI Actions, we just need to move them to UI Actions plugin/package.
- state containers - an alternative to Redux, SharedUx could figure out what to do with it next.

---

All the navigation, `history`, routing logic is owned by Core team and application teams build on top of it. So, the future of all below items need to be decided between the Core team and the teams who use them Data/Discover/Presentation/Viz. All below are additions to the history/url/routing logic, which is at the base level provided by core and apps teams use these helpers on top:

- history
- state management
- state sync
- storage
- state management & url helpers

---

A couple of items directly related to Core:

- core (createStartServicesGetter) - synchronous way to access "start" contract.
- persistable state - deals with migrations, Stacey and Peter want to put it in Core from the beginning.

---

Various other things, probably used by AX team:

- resize checker
- KbnServerError & related utilities
- KbnError and other error utilities
- a handful of other small utilities which appear largely used by data/discover plugins

Reply:
Yes.  The plugin makes no sense with current teams mandates.  @vadimkibana - let's own this split up and get this update merged.  Please create a meta issue to track the split up activity.

Reply:
I've created PR[here](https://github.com/elastic/kibana/pull/168209), which assigns everything to SharedUX temporarily, lets discuss in team how we want to proceed further, be it meta issue or any other mechanism.